process/smartContract/builtInFunctions/factory.go

```diff
@@ -13,44 +13,44 @@ import (
 
 // ArgsCreateBuiltInFunctionContainer defines the argument structure to create new built in function container
 type ArgsCreateBuiltInFunctionContainer struct {
-	GasSchedule                          core.GasScheduleNotifier
-	MapDNSAddresses                      map[string]struct{}
-	EnableUserNameChange                 bool
-	Marshalizer                          marshal.Marshalizer
-	Accounts                             state.AccountsAdapter
-	ShardCoordinator                     sharding.Coordinator
-	EpochNotifier                        vmcommon.EpochNotifier
-	ESDTMultiTransferEnableEpoch         uint32
-	ESDTTransferRoleEnableEpoch          uint32
-	GlobalMintBurnDisableEpoch           uint32
-	ESDTTransferMetaEnableEpoch          uint32
-	ESDTNFTCreateOnMultiShardEnableEpoch uint32
+	GasSchedule                  core.GasScheduleNotifier
+	MapDNSAddresses              map[string]struct{}
+	EnableUserNameChange         bool
+	Marshalizer                  marshal.Marshalizer
+	Accounts                     state.AccountsAdapter
+	ShardCoordinator             sharding.Coordinator
+	EpochNotifier                vmcommon.EpochNotifier
+	ESDTMultiTransferEnableEpoch uint32
+	ESDTTransferRoleEnableEpoch  uint32
+	GlobalMintBurnDisableEpoch   uint32
+	ESDTTransferMetaEnableEpoch  uint32
+	OptimizeNFTStoreEnableEpoch  uint32
 }
 
 // CreateBuiltInFunctionContainer creates a container that will hold all the available built in functions
-func CreateBuiltInFunctionContainer(args ArgsCreateBuiltInFunctionContainer) (vmcommon.BuiltInFunctionContainer, error) {
+func CreateBuiltInFunctionContainer(args ArgsCreateBuiltInFunctionContainer) (vmcommon.BuiltInFunctionContainer, vmcommon.SimpleESDTNFTStorageHandler, error) {
```

Comment:
maybe add a todo to refactor this:
this factory should return the `bContainerFactory` and on caller side you should get the container and the nft storage handler

Reply:
renamed.